py/sigver/test/test_microsign.py

```diff
@@ -0,0 +1,70 @@
+# SPDX-FileCopyrightText: 2023 Technology Innovation Institute (TII)
+# SPDX-License-Identifier: Apache-2.0
+""" Signature Verification Microservice Tests """
+import subprocess
+import time
+import pytest
+
+# Constants for production and staging URLs
+STAGING_SLOT_URL = "https://ghaf-devenv-microsign-aleksandrtserepo-app.azurewebsites.net/api/verify-signature" # pylint: disable=line-too-long
+PRODUCTION_SLOT_URL = "https://ghaf-devenv-signverify.azurewebsites.net/api/verifysignature"
+
+# Test configuration
+MAX_RETRIES = 3  # Max number of retries in case of failure
+SLEEP_INTERVAL = 5  # Time to wait between retries or requests (seconds)
+
+@pytest.fixture(scope="session")
+def retry_config():
+    """ Fixture to define the maximum retries and sleep interval """
+    return {
+        'max_retries': MAX_RETRIES,
+        'sleep_interval': SLEEP_INTERVAL
+    }
+
+def run_verification_script(url):
+    """ Function to run step.py script with --url parameter """
+    try:
+        result = subprocess.run(['python3', 'step.py', f'--url={url}'],
+                                capture_output=True, check=True)
+        return result.returncode
+    except subprocess.CalledProcessError as e:
+        print(f"Exception while running step.py for {url}: {e}")
+        return -2  # Return code for exception
+
+def run_test_with_retries(url, max_retries, sleep_interval):
+    """  Function to run verification with retry logic """
+    retry_count = 0
+    while retry_count < max_retries:
+        result_code = run_verification_script(url)
+        if result_code == 0:
+            print(f"Test passed for {url}.")
+            return True
+        if result_code == -5:
+            print(f"Received 503 from {url}. Retrying... (retry count: {retry_count + 1})")
+            retry_count += 1
+            time.sleep(sleep_interval)
+        else:
```

Comment:
Should result code 1(?) indicating that signature verification process seems to work, but signature is wrong, be handled differently from a complete failure?

Reply:
Good point. The thing was that we were trying to catch 503 here as that was the biggest concern (intermittent 503 most likely due to cold start) and I didn't pay too much attention at the other details. Will fix. Thanks!